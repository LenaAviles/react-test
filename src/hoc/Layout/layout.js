import React from 'react'
import PropTypes from "prop-types";

import AppHeader from '../../components/Header';
import SideMenu from '../../components/SideMenu';
import NavBar from '../../components/NavBar';
import './layout.css';

const Layout = (props) => {
    return (
        <div>
            <AppHeader />
            <NavBar />
            <SideMenu />
            {props.children}
        </div>
    )
}

Layout.propTypes = {
    children: PropTypes.element.isRequired,
}

export default Layout;
