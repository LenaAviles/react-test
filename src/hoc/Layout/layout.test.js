import React from 'react';
import Layout from './layout';
import { Switch } from 'react-router-dom';
import ShallowRenderer  from 'react-test-renderer/shallow';

jest.mock('react-router-dom', () => ({ Switch: 'Switch' }))

const mockProps = { children: <Switch />};

it('renders Layout correctly', () => {
  const renderer = new ShallowRenderer()  
  const tree = renderer
    .render(<Layout {...mockProps}/>);
  expect(tree).toMatchSnapshot();
});