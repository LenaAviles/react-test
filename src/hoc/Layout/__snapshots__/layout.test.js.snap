// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Layout correctly 1`] = `
<div>
  <Connect(Header) />
  <Connect(NavBar) />
  <Connect(SideMenu) />
  <Switch />
</div>
`;
